### ISSUE TITLE
arm64

### ISSUE THREAD

**Issue by simonSlamka** [ActivityWatch/activitywatch#641]
are you planning on releasing arm64 builds? would be very useful since that's where computers are headed (at least laptops) and some don't have anything else available

- [x] I have searched the [issues](https://github.com/ActivityWatch/activitywatch/issues) of this repo and believe that this is not a duplicate.
- [x] I have searched the [documentation](https://docs.activitywatch.net/en/latest/) and believe that my question is not covered.

## Issue

same as above

**Comment by johan-bjareholt** [comment]
I have compiled aw-server-rust and aw-watcher-window-wayland for my Librem 5 Linux phone and have also tried it on a PinePhone, so at least it partially works.

The issue I think is rather to know which platforms are worth supporting ARM for, as it means more maintenance. It's also a question about which platforms our CI works well with (GitHub Actions).

**Comment by simonSlamka** [comment]
> The issue I think is rather to know which platforms are worth supporting ARM for

imo, Linux, for sure. Then definitely Darwin since Silicons will be gradually more and more popular and widespread. Additionally, it's likely that more manufacturers will decide to put arm64's into their laptops because of its many upsides

**Comment by lacamar** [comment]
With Asahi linux bringing linux to ARM, we are at an inflection point for ARM64 adoption. It's going to be increasingly important for software to work on ARM64 hardware.

**Comment by simonSlamka** [comment]
> With Asahi linux bringing linux to ARM, we are at an inflection point for ARM64 adoption. It's going to be increasingly important for software to work on ARM64 hardware.

Exactly this! +1

**Comment by cyugao** [comment]
Any progress on this? It should work for Apple Silicon Macs. I compiled it on my Mac and it's good for use so far.

**Comment by ErikBjare** [comment]
Can't make progress on this until someone figures out how to build arm64
builds on GitHub CI (which only offers x86 machines).

This is low-priority for me as it seems the current build works just fine,
and getting arm64 to build *feels* like it'll be a lot of work (unless you
already know how to, unlike me).

**Comment by yeradis** [comment]
Haven't tested it yet but found this info https://github.com/actions/runner-images/issues/2187#issuecomment-1295921395
regarding a service with on-demand Apple Silicon runners for GitHub Actions.

**Comment by jbg** [comment]
Arrived here after checking out the website, deciding to install, then realising there is no arm64 macOS build.

Which languages are used here? Cross-compiling Rust is very easy (`rustup target add aarch64-apple-darwin && cargo build --target aarch64-apple-darwin`) and Python usually doesn't require anything special either. There's no need to *build on* Apple Silicon just to *target* Apple Silicon. I can submit a PR to add Apple Silicon support, just need to look over the repo (or get some pointers) to understand which languages are used other than the Rust and Python that I've already seen in server-rust and watcher-window.

**Comment by johan-bjareholt** [comment]
@jbg Cross-compiling Rust to ARM64 is easy, so that's not a big issue. While running python on ARM64 is easy, using pyinstaller to bundle everything is not as easy. It seems like pyinstaller has gotten decent ARM64 the past few years, but since GitHub Actions don't support macOS ARM runners we also need to be able to cross-compile ARM64 with pyinstaller which I'm sure is a bit more tricky. An option to cross-compiling would be to build in a virtualized ARM64 environment which @yeradis mentioned which might work.

Feel free to dig further, the short answer is really that no one has given this an attempt.

**Comment by jlumbroso** [comment]
It seems https://cirrus-ci.org/pricing/ indicates free service for public open source projects.

I am willing to sponsor whoever implements the Apple Silicon build of ActivityWatch. ActivityWatch is a killer app and it would be great to have it available natively, given it runs in the background.

**Comment by allywilson** [comment]
+1 for aarch64 (using Fedora Asahi).

**Comment by imadnyc** [comment]
+1 for aarch64 (on mobian)

**Comment by thomasjonas** [comment]
It seems [Github has Apple Silicon runners in beta](https://github.blog/2023-10-02-introducing-the-new-apple-silicon-powered-m1-macos-larger-runner-for-github-actions/). That should make it possible to build for Apple Silicon for sure, not sure if that means it would also work for Fedora Asahi or other arm builds though...

Would it be possible to use the Apple Silicon beta runner? Would that mean the workflow/build has to be adjusted?

**Comment by wojnilowicz** [comment]
> +1 for aarch64 (using Fedora Asahi).

You can test it from https://copr.fedorainfracloud.org/coprs/wojnilowicz/activitywatch/

**Comment by FloLecoeuche** [comment]
Are there any sticking points today?

**Comment by rtpHarry** [comment]
Oh no. I have made a grave mistake. I finally bought the M4 and I didn't check first if this was supported. Hoping that it will get native support soon and doesn't gobble too many resources using whatever the compatibility mode is...

**Comment by BelKed** [comment]
Don't worry, I have ActivityWatch running on my M1 and it works quietly in the background without any errors :)

**Comment by dingwen07** [comment]
+1 for Windows Arm64

**Comment by 0xbrayo** [comment]
> +1 for Windows Arm64

Currently not possible, github does not provide Windows arm runners. It runs via emulation though, should be smooth. The runners could be available before next release will keep you posted.

**Comment by rtpHarry** [comment]
> Don't worry, I have ActivityWatch running on my M1 and it works quietly in the background without any errors :)

Yeah I am running it now on the M4 and its just as good as it always has been. I don't know if maybe its using more resources than it would, but its not make a scratch on this laptop.

**Comment by rggjan** [comment]
+1 for native macOS support. This is pretty much the only application still using Rosetta on macOS for me.

**Comment by 0xbrayo** [comment]
[aw-tauri](https://github.com/ActivityWatch/aw-tauri) will be compiled for arm64. It's not ready yet, but will be soon.

**Comment by rs38** [comment]
> > +1 for Windows Arm64
> 
> Currently not possible, github does not provide Windows arm runners. It runs via emulation though, should be smooth. The runners could be available before next release will keep you posted.

now already avail since April:
https://github.blog/changelog/2025-04-14-windows-arm64-hosted-runners-now-available-in-public-preview/

**Comment by NathanUrwin** [comment]
> +1 for aarch64 (using Fedora Asahi).

same

**Comment by wojnilowicz** [comment]
On 25-12-28 21:31, Nathan Urwin wrote:
>NathanUrwin left a comment (ActivityWatch/activitywatch#641)
>
>> +1 for aarch64 (using Fedora Asahi).
>
>same
>

You can install it already by `sudo dnf in aw-server-rust python3-aw-qt`

**Comment by rggjan** [comment]
https://docs.github.com/en/actions/reference/runners/github-hosted-runners

Looks like there's arm64 based runners for macos available as well now?

**Comment by micxer** [comment]
Why is a windows runner needed to build an arm version for native macOS? Really looking forward to see this as a native app. I would also need to install Rosetta only for this and would love ot use it to track myself throughout the day.

Is there any way to put it together by myself somehow?

**Comment by 0xbrayo** [comment]
> Why is a windows runner needed to build an arm version for native macOS?

Not sure where you got that from. It's not necessary and the next version will feature macos arm app. Some work still needs to be done for windows, not all crates are arm compatible on windows

**Comment by micxer** [comment]
My bad. I had macOS arm in my head and read https://github.com/ActivityWatch/activitywatch/issues/641#issuecomment-2684025609. But yeah, happy to hear that arm is on the way.
